os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -434,42 +437,95 @@ static void _up_assert(int errorcode)
 }
 
 #ifdef CONFIG_BINMGR_RECOVERY
+/* Recover semaphores and watchdog timer resources */
+static void recovery_task_recover(struct tcb_s *tcb)
```

Comment:
1. Let's name it like `<module_name>_<verb>_<obj>`.
2. I think it looks strange of function location. Actually this is not assert code.